File: MySpringApp_v2/src/main/java/com/nikolaev/review/ReviewService.java
package com.nikolaev.review;

import com.nikolaev.review.dto.ReviewDto;

public interface ReviewService {
    ReviewDto getReview(Long id);

    ReviewDto update(ReviewDto reviewDto);

    ReviewDto submitReview(Long reviewId);

    void deleteReview(Long reviewId);

    void createReview(Long documentId, ReviewDto reviewDto);
}

File: my-app/src/app/core/service/review.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs/Observable';
import {Review} from '../../shared/model/review/review.model';

@Injectable()
export class ReviewService {
  constructor(private http: HttpClient) {
  }

  getReview(reviewId: number): Observable<Review> {
    return this.http.get<Review>('/api/reviews/' + reviewId);
  }

  createReview(documentId: number, review: Review) {
    return this.http.post('/api/documents/' + documentId + '/reviews', review);
  }


  updateReview(review: Review): Observable<Review> {
    return this.http.put<Review>('/api/reviews', review);
  }

  submitReview(review: Review): Observable<Review> {
    return this.http.post<Review>('/api/reviews/' + review.id + '/submit', review);
  }

  deleteReview(reviewId: number) {
    return this.http.delete('/api/reviews/' + reviewId);
  }
}
